apps/test/unit/code-studio/pd/application/TeacherApplicationTest.js

```diff
@@ -114,24 +116,28 @@ describe('TeacherApplication', () => {
       ).to.deep.equal({school: '16'});
     });
     it('has only saved form data and no school id if session storage has school info', () => {
-      // [MEG] TODO: Use FakeStorage instead
-      window.sessionStorage.setItem(
-        'TeacherApplication',
-        JSON.stringify({data: {school: '25'}})
-      );
-      teacherApplication.mergeProps({savedFormData, schoolId});
+      let getItem = sinon.stub(storage, 'getItem');
+      getItem
+        .withArgs('TeacherApplication')
+        .returns(JSON.stringify({data: {school: '25'}}));
```

Comment:
this could be made simpler by taking advantage of the work that's already been put into FakeStorage. you've already defined `const storage = new FakeStorage()`, so this can just be:
```suggestion
      storage.setItem('TeacherApplication', JSON.stringify({data: {school: '25'}})
```

Reply:
Unless I'm missing something, it looks like [FakeStorage](https://github.com/code-dot-org/code-dot-org/blob/c3325655902e82479d0a85d5adc73049810e5b66/apps/test/util/FakeStorage.js) doesn't actually adjust getItem based on setItem ... when I try that on the test, `getItem` isn't retrieving anything

Reply:
With @tim-dot-org's help, was able to stub out window.sessionStorage without the use of FakeStorage

Reply:
> Unless I'm missing something, it looks like [FakeStorage](https://github.com/code-dot-org/code-dot-org/blob/c3325655902e82479d0a85d5adc73049810e5b66/apps/test/util/FakeStorage.js) doesn't actually adjust `getItem` based on `setItem` ... when I try that on the test, `getItem` isn't retrieving anything

oh jeez 🤦‍♂️   you're right. well something named fake storage really SHOULD have a fake implementation! :-) https://martinfowler.com/bliki/TestDouble.html but no need to build that now.

Reply:
Lol well it feels validating to know there's not secret magic happening